cli/src/main/kotlin/Main.kt

```diff
@@ -0,0 +1,535 @@
+                names = ["--allow-dynamic-versions"],
+                order = PARAMETER_ORDER_OPTIONAL)
+        private var allowDynamicVersions = false
+
+        @Parameter(description = "By default the components configured to be excluded in the .ort.yml file of the " +
+                "input directory will still be analyzed, but marked as excluded in the result. With this option " +
+                "enabled they will not be analyzed and not appear in the result.",
+                names = ["--remove-excludes-from-result"],
+                order = PARAMETER_ORDER_OPTIONAL)
+        private var removeExcludesFromResult = false
+
+        @Parameter(description = "A YAML file that contains package curation data.",
+                names = ["--package-curations-file"],
+                order = PARAMETER_ORDER_OPTIONAL)
+        private var packageCurationsFile: File? = null
+
+        override fun run() {
+            val absoluteOutputPath = outputDir.absoluteFile
+            if (absoluteOutputPath.exists()) {
+                log.error { "The output directory '$absoluteOutputPath' must not exist yet." }
+                exitProcess(1)
+            }
+
+            println("The following package managers are activated:")
+            println("\t" + packageManagers.joinToString(", "))
+
+            val absoluteProjectPath = inputDir.absoluteFile
+            println("Scanning project path:\n\t$absoluteProjectPath")
+
+            val config = AnalyzerConfiguration(ignoreToolVersions, allowDynamicVersions, removeExcludesFromResult)
+            val analyzer = Analyzer(config)
+            val ortResult = analyzer.analyze(absoluteProjectPath, packageManagers, packageCurationsFile)
+
+            absoluteOutputPath.safeMkdirs()
+
+            outputFormats.forEach { format ->
+                val outputFile = File(absoluteOutputPath, "analyzer-result.${format.fileExtension}")
+                println("Writing analyzer result to '$outputFile'.")
+                format.mapper.writerWithDefaultPrettyPrinter().writeValue(outputFile, ortResult)
+            }
+        }
+    }
+
+    @Parameters(commandNames = ["download"], commandDescription = "Fetch source code from a remote location.")
+    private object DownloaderCommand : Runnable {
+        @Parameter(description = "A dependencies analysis file to use. Must not be used together with '--project-url'.",
+                names = ["--dependencies-file", "-d"],
+                order = PARAMETER_ORDER_OPTIONAL)
+        private var dependenciesFile: File? = null
+
+        @Parameter(description = "A VCS or archive URL of a project to download. Must not be used together with " +
+                "'--dependencies-file'.",
+                names = ["--project-url"],
+                order = PARAMETER_ORDER_OPTIONAL)
+        private var projectUrl: String? = null
+
+        @Parameter(description = "The speaking name of the project to download. Will be ignored if " +
```

Comment:
Maybe add a hint that this should be used together with `--projectUrl` instead of only saying to not use it with `--dependencies-file`, as below.